src/config.ts

```diff
@@ -0,0 +1,72 @@
+const SERVICE_NAME = "auth-service";
+
+const {
+    NATS_URL = "nats://localhost:4222",
+    PROTOCOL = "https",
+    OKTA_ORG = "playtix.okta.com",
+    OKTA_AUTHORIZATION_SERVER = "default",
+    OKTA_API_VERSION = "v1",
+    INTERNAL_REDIRECT_URI = "http://localhost:8000/auth/auth-service/authorize_code"
+} = process.env;
+
+const HANDLERS_SUBJECTS = {
+    TOKEN: "token",
+    AUTHN: "authn",
+    SESSION: "sessions",
+    USERINFO: "userinfo",
+    AUTHORIZE: "authorize",
+    AUTHORIZE_CODE: "authorize_code"
+};
+
+if (!NATS_URL) {
+    throw new Error(
+        `Please provide a valid NATS_URL so the service can connect to NATS. 
+        For example, use nats://nats:4222`
+    );
+}
+
+if (!PROTOCOL) {
+    throw new Error(
+        `Please provide a valid protocol. For example, http or https`
+    );
+}
+
+if (!OKTA_ORG) {
+    throw new Error(
+        `Please provide a valid Okta organization. For example, abcd.okta.com`
+    );
+}
+
+if (!OKTA_AUTHORIZATION_SERVER) {
+    throw new Error(
+        `Please provide a valid Okta authorization server. For example, default`
+    );
+}
+
+if (!OKTA_API_VERSION) {
+    console.warn(
+        "Okta API not specified (OKTA_API_VERSION) using the v1 as default value."
+    );
+}
+
+const OKTA_URI = `${PROTOCOL}://${OKTA_ORG}`;
```

Comment:
should the var names have Okta in them? is the flow specific to okta or common to all oauth2 services? if the auth flow works with all oauth2 services I'd suggest removing OKTA?

Reply:
Indeed, the flow is common to all OAuth2.0 services. I'll remove OKTA and also ask for the links to be defined in the environment in order to avoid any url building error.